### DISCUSSION TITLE
Incorrect blockchain: arbitrum-mainnet.gateway.pokt.network

### DISCUSSION THREAD

**Discussion by ghost** [WalletConnect/walletconnect-monorepo#2674] [Developer Support]
RPC calls are being made to:  `https://rpc.walletconnect.com/v1/?chainId=eip155:42161&projectId=xxx`

75% of them get this error message:
```
{
    "jsonrpc": "2.0",
    "id": 6,
    "error": {
        "message": "Incorrect blockchain: arbitrum-mainnet.gateway.pokt.network",
        "code": -32057
    }
}
```

Retrying requests occasionally works.  The error is transient, but it's making Arbitrum apps nearly completely unusable.

To me, it looks like walletconnect is using pokt.network and hitting a bad node.

Is there any way to control which RPC's walletconnect ultimately uses?

**Comment by arein** [comment]
Thank you for the report. We will follow up shortly.

**Comment by xzilja** [comment]
User provided more info in this issue: https://github.com/WalletConnect/web3modal/issues/1078

**Comment by arein** [comment]
@reupcash confirming bug (typo..) on our end. Tests didn't catch this as we don't test all providers for each chain.

Will be rolled out by end of day tomorrow. Thanks for the bug report!

**Reply by startswithaj** [reply]
@arein can we configure our own RPC?

**Reply by arein** [reply]
Yea of course! Check Web3modal docs. We provide an RPC as convenience not to force anybody unto us @startswithaj

**Reply by ghost** [reply]
We call `configureChains` with two providers:  `w3mProvider` and `jsonRpcProvider`

But it did not help us at all in this scenario

PS:  The `readContract` function threw an error (exception), as opposed to returning a result object with an error property set.  Not sure if that makes any difference

**Reply by arein** [reply]
@crypblizz8 can you pls help?

**Comment by arein** [comment]
Arbitrum RPC fix has been rolled out

**Reply by ghost** [reply]
Thanks for the super-speedy action on this

**Reply by arein** [reply]
No. Thank you for reporting!

**Reply by layinka** [reply]
I am getting this error on goerli as at today

**Comment by Danikovvv** [comment]
thanks, its good

**Reply by Nniikka** [reply]
Good job guys!

**Reply by syifamutia7** [reply]
Yes, good job!
